### PR TITLE
`sage.schemes`: Modularization fixes (imports)

### PR DESCRIPTION
**Pull request by mkoeppe**
- Cherry-picked from #35095.

### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/schemes/elliptic_curves/ell_curve_isogeny.py [hunk 1]
```diff
@@ -3719,6 +3719,7 @@ def compute_intermediate_curves(E1, E2):
     EXAMPLES::
 
         sage: from sage.schemes.elliptic_curves.ell_curve_isogeny import compute_intermediate_curves
+
         sage: E = EllipticCurve(GF(83), [1,0,1,1,0])
```
Comment (kwankyu) [inline src/sage/schemes/elliptic_curves/ell_curve_isogeny.py]: Why space here?

As far as I know, this is not a convention.

File: src/sage/schemes/hyperelliptic_curves/hyperelliptic_generic.py [hunk 2]
```diff
@@ -38,7 +38,8 @@
 from sage.rings.power_series_ring import PowerSeriesRing
 from sage.rings.laurent_series_ring import LaurentSeriesRing
 from sage.rings.real_mpfr import RR
-from sage.functions.all import log
+from sage.misc.lazy_import import lazy_import
+lazy_import("sage.functions.all", "log")
 from sage.structure.category_object import normalize_names
```
Comment (kwankyu) [inline src/sage/schemes/hyperelliptic_curves/hyperelliptic_generic.py:44]: How about grouping `lazy_import` statement(s) separated from normal imports, as done in other files?

File: src/sage/schemes/toric/toric_subscheme.py [hunk 3]
```diff
@@ -21,7 +21,8 @@
 # https://www.gnu.org/licenses/
 # ****************************************************************************
 
-from sage.calculus.functions import jacobian
+from sage.misc.lazy_import import lazy_import
+lazy_import("sage.calculus.functions", "jacobian")
 from sage.rings.integer_ring import ZZ
```
Comment (kwankyu) [inline src/sage/schemes/toric/toric_subscheme.py:24]: Separate out `lazy_import` statement?

File: src/sage/schemes/toric/variety.py [hunk 4]
```diff
@@ -302,7 +304,8 @@
 
 import sys
 
-from sage.functions.all import factorial
+from sage.misc.lazy_import import lazy_import
+lazy_import("sage.functions.all", "factorial")
 import sage.geometry.abc
```
Comment (kwankyu) [inline src/sage/schemes/toric/variety.py:309]: The same here.

**Comment by kwankyu** [comment]
Otherwise lgtm.

File: src/sage/schemes/elliptic_curves/ell_curve_isogeny.py (hunk 1 above)
Comment (mkoeppe) [reply src/sage/schemes/elliptic_curves/ell_curve_isogeny.py]: This change was a leftover from marking the following block with a `# needs` that is not needed any more. 
I've removed the blank line again in 2a3f2bf5258

File: src/sage/schemes/hyperelliptic_curves/hyperelliptic_generic.py (hunk 2 above)
Comment (mkoeppe) [reply src/sage/schemes/hyperelliptic_curves/hyperelliptic_generic.py:44]: Thanks, done in f52c4fe8c5e

**Review by kwankyu — APPROVED** [review]
Thanks.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**